File: app/src/main/java/com/example/schoolairdroprefactoredition/fragment/settings/SettingsGeneralFragment.java
package com.example.schoolairdroprefactoredition.fragment.settings;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;

import com.example.schoolairdroprefactoredition.R;
import com.example.schoolairdroprefactoredition.fragment.TransactionBaseFragment;

/**
 * 通用
 */
public class SettingsGeneralFragment extends TransactionBaseFragment implements View.OnClickListener {

    private String languageName;

    private FragmentManager manager;

    public SettingsGeneralFragment() {
    }

    public static SettingsGeneralFragment newInstance(Bundle bundle) {
        SettingsGeneralFragment fragment = new SettingsGeneralFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        languageName = getResources().getString(R.string.language);

        if (getActivity() != null)
            manager = getActivity().getSupportFragmentManager();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        final View root = LayoutInflater.from(getContext()).inflate(R.layout.fragment_settings_general, container, false);

        root.findViewById(R.id.settings_general_language).setOnClickListener(this);
        root.findViewById(R.id.settings_general_storage).setOnClickListener(this);

        return root;
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.settings_general_language)
            transact(manager, new SettingsLanguageFragment(), languageName);
    }
}

File: app/src/main/java/com/example/schoolairdroprefactoredition/scene/settings/fragment/SettingsGeneralFragment.kt
package com.example.schoolairdroprefactoredition.scene.settings.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.schoolairdroprefactoredition.application.SAApplication
import com.example.schoolairdroprefactoredition.R
import com.example.schoolairdroprefactoredition.cache.util.UserSettingsCacheUtil
import com.example.schoolairdroprefactoredition.scene.base.TransitionBaseFragment
import com.example.schoolairdroprefactoredition.ui.components.PageItem
import kotlinx.android.synthetic.main.fragment_settings_general.*

class SettingsGeneralFragment : TransitionBaseFragment(),
        View.OnClickListener, PageItem.OnPageItemSelectedListener {

    /**
     * 暗黑模式按钮是否已经初始化
     */
    private var isDarkThemeInitialized = false

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return LayoutInflater.from(context).inflate(R.layout.fragment_settings_general, container, false).rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (UserSettingsCacheUtil.getInstance().isDarkTheme()) {
            settingsGeneralDarkTheme.select()
        }
        isDarkThemeInitialized = true

        settingsGeneralStorage.setOnClickListener(this@SettingsGeneralFragment)
        settingsGeneralDarkTheme.setOnClickListener(this@SettingsGeneralFragment)
        settingsGeneralDarkTheme.setOnPageItemSelectedListener(this@SettingsGeneralFragment)
    }

    override fun onClick(v: View?) {
        when (v) {
            settingsGeneralStorage -> {
            }

            settingsGeneralDarkTheme -> {
                settingsGeneralDarkTheme.toggle()
            }
        }
    }

    override fun onPageItemToggled(pageItem: PageItem?) {
        when (pageItem) {
            settingsGeneralDarkTheme -> {
                if (isDarkThemeInitialized) {
                    if (activity?.application is SAApplication) {
                        (activity?.application as SAApplication).setAppTheme(settingsGeneralDarkTheme.isItemSelected)
                    }
                    activity?.supportFragmentManager?.popBackStack()
                }
            }
        }
    }
}